### DISCUSSION TITLE
Update or new version?

### DISCUSSION THREAD

**Discussion by aalmans** [mrhan1993/Fooocus-API#381] [General]
HI,

I was planning to update my Fooocus-api instance as I am still running 0.4.0.7 but I also noticed that there is a new Fooocus-api build? Which one should I use going forward and is there a way to perform automatic updates? Or do I just need to pull a new version and install over the old one? Thanks!

**Comment by mrhan1993** [comment]
If I understand correctly, you're wondering whether you should update to my refactored project, which is FooocusAPI. 

One way to decide is to consider whether you can accept significant changes in the interfaces. The most notable feature of the new project is that all generated interfaces are centralized.

If you cannot accept this, you can continue to use this project and simply update with git pull.

If you don't mind this change, then I suggest you first take a look at the documentation of the new project, as there are changes in the request parameters, return methods, and return formats.

If my understanding is incorrect, or if you have any other questions, please let me know.

**Reply by aalmans** [reply]
Hi there, i am not too familiar with the git update command. I am running Fooocus-api from anaconda. What would the command be from there to update? Thanks!

**Reply by mrhan1993** [reply]
Open the terminal in the root directory of your project and execute `git pull`.

**Reply by aalmans** [reply]
Thanks that works. You may want to add this to the documentation for people who are also not familiar with this process!

**Reply by aalmans** [reply]
Hmm after upgrading from 04.0.7 i get the following error message:

[2024-07-11 12:26:21] ERROR    [Fooocus] Worker error: probability tensor contains either `inf`, `nan` or element < 0
[2024-07-11 12:26:21] INFO     [Task Queue] Finish task with error, job_id=b2c1ea21-d29a-43de-a2d5-0799b106ce51

What could be the case?

**Reply by mrhan1993** [reply]
> Hmm after upgrading from 04.0.7 i get the following error message:
> 
> [2024-07-11 12:26:21] ERROR    [Fooocus] Worker error: probability tensor contains either `inf`, `nan` or element < 0
> [2024-07-11 12:26:21] INFO     [Task Queue] Finish task with error, job_id=b2c1ea21-d29a-43de-a2d5-0799b106ce51
> 
> What could be the case?

What are your request parameters?

**Reply by aalmans** [reply]
Never mind. A restart of my computer seems to have fixed it!
